include/tachimawari/control/sdk/module/dynamixel_sdk.hpp

```diff
@@ -26,65 +26,47 @@
 #include <string>
 #include <vector>
 
+#include "dynamixel_sdk/dynamixel_sdk.h"
 #include "tachimawari/control/manager/control_manager.hpp"
 #include "tachimawari/control/sdk/packet/model/group_bulk_read.hpp"
 #include "tachimawari/control/sdk/packet/protocol_1/group_bulk_read.hpp"
 #include "tachimawari/control/sdk/packet/protocol_1/group_sync_write.hpp"
 
-#include "dynamixel_sdk/dynamixel_sdk.h"
-
 namespace tachimawari::control
 {
 
 class DynamixelSDK : public ControlManager
 {
 public:
-  enum : int
-  {
-    SUCCESS = 0,
-    TX_FAIL = -1001
-  };
+  enum : int { SUCCESS = 0, TX_FAIL = -1001 };
 
   explicit DynamixelSDK(
-    const std::string & port_name, int baudrate = 1000000,
-    float protocol_version = 1.0);
+    const std::string & port_name, int baudrate = 1000000, float protocol_version = 1.0);
   ~DynamixelSDK();
 
   void set_port(const std::string & port_name) override;
 
-
   bool connect() override;
   void disconnect() override;
 
   bool ping(uint8_t id) override;
 
-  bool write_packet(
-    uint8_t id, uint8_t address, int value,
-    int data_length = 1) override;
+  bool write_packet(uint8_t id, uint16_t address, int value, int data_length = 1) override;
 
-  int read_packet(
-    uint8_t id, uint8_t address, int data_length = 1) override;
+  int read_packet(uint8_t id, uint16_t address, int data_length = 1) override;
 
-  bool sync_write_packet(
-    const std::vector<joint::Joint> & joints,
-    bool with_pid = false) override;
+  bool sync_write_packet(const std::vector<joint::Joint> & joints, bool with_pid = false) override;
 
-  bool bulk_read_packet() override;
-
-  int get_data(
-    uint8_t id, uint16_t address, int data_lenghth = 1);
-
-  int get_bulk_data(
-    uint8_t id, uint8_t address,
-    int data_length = 1) override;
+  bool send_bulk_read_packet() override;
+  bool add_default_bulk_read_packet() override;
+  int get_data(uint8_t id, uint16_t address, int data_lenghth = 1) override;
+  int get_bulk_data(uint8_t id, uint16_t address, int data_length = 1) override;
 
 private:
-  bool send_bulk_read_packet();
-
   dynamixel::PortHandler * port_handler;
   dynamixel::PacketHandler * packet_handler;
 
-  std::shared_ptr<std::map<uint8_t, std::shared_ptr<sdk::GroupBulkRead>>> bulk_data;
+  std::shared_ptr<std::unordered_map<uint8_t, std::vector<uint8_t>>> bulk_data;
```

Comment:
I just checked the code, and i see that we don't actually need the `bulk_read` to be in `std::shared_ptr`. We can just using a normal `std::unordered_map`:

```suggestion
  std::unordered_map<uint8_t, std::vector<uint8_t>> bulk_data;
```

And pass it to other functions (e.g. `GroupBulkRead::insert_all`) as reference:

```c++
static void insert_all(
  std::unordered_map<uint8_t, std::vector<uint8_t>> & bulk_data,
  const std::shared_ptr<sdk::GroupBulkRead> & sdk_group_bulk_read);
```

But let's do that in other PR, after this one merged.

Reply:
Okay mas fi, thanks for ur suggestions.